Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractRandomCutTree.java

```diff
@@ -216,6 +218,33 @@ abstract protected NodeReference addNode(NodeReference parent, NodeReference lef
 
     abstract protected void deleteSequenceIndex(NodeReference node, long uniqueSequenceNumber);
 
+    abstract boolean modifyBoxAndCheckContains(NodeReference tempNode, Point point);
+
+    abstract void readjustPointSum(NodeReference node, Point point);
+
+    void recursivelyReadjustPointSum(NodeReference node, Point point) {
+        if (enableCenterOfMass) {
+            NodeReference tempNode = node;
+            while (tempNode != null) {
+                readjustPointSum(tempNode, point);
+                tempNode = getParent(tempNode);
+            }
+        }
+    }
+
+    // manages the bounding boxes and center of mass
+    void updateAncestorNodesAfterDelete(NodeReference nodeReference, Point point) {
+        NodeReference tempNode = nodeReference;
+        boolean boxNeedsUpdate = enableCache;
+        while (boxNeedsUpdate && tempNode != null) {
+            boxNeedsUpdate = !modifyBoxAndCheckContains(tempNode, point);
+            tempNode = getParent(tempNode);
+        }
+
+        recursivelyReadjustPointSum(nodeReference, point);
```

Comment:
See, we have "update" and "modify" and "adjust" all in this one class. I'm in favor of using descriptive names, but to me those three are too similar to justify using all three. "Readjust" is better since it indicates that we're doing something again, but I think I prefer "recompute" because it suggests more strongly that there's a correct result.

Reply:
renamed to reComputePointSum